ur_controllers/src/scaled_joint_trajectory_controller.cpp

```diff
@@ -87,189 +87,222 @@ controller_interface::return_type ScaledJointTrajectoryController::update(const
     return controller_interface::return_type::OK;
   }
 
-  auto resize_joint_trajectory_point = [&](trajectory_msgs::msg::JointTrajectoryPoint& point, size_t size) {
-    point.positions.resize(size);
-    if (has_velocity_state_interface_) {
-      point.velocities.resize(size);
-    }
-    if (has_acceleration_state_interface_) {
-      point.accelerations.resize(size);
-    }
-  };
-  auto compute_error_for_joint = [&](JointTrajectoryPoint& error, int index, const JointTrajectoryPoint& current,
+  auto compute_error_for_joint = [&](JointTrajectoryPoint& error, size_t index, const JointTrajectoryPoint& current,
                                      const JointTrajectoryPoint& desired) {
     // error defined as the difference between current and desired
-    error.positions[index] = angles::shortest_angular_distance(current.positions[index], desired.positions[index]);
-    if (has_velocity_state_interface_ && has_velocity_command_interface_) {
+    if (normalize_joint_error_[index]) {
+      // if desired, the shortest_angular_distance is calculated, i.e., the error is
+      //  normalized between -pi<error<pi
+      error.positions[index] = angles::shortest_angular_distance(current.positions[index], desired.positions[index]);
```

Comment:
That particular line is (as most of the update function) from the upstream JTC. This is used only if

- closed loop control is activated.
- `angle_wraparound` is set to `true` in the controller configuration.

Both isn't true inside our controller configuration and also has to be explicitly activated when desired. I would vote for sticking as close to the upstream code as possible because

a) we want to make our lives easy to adapt to future changes
b) we want to switch to the upstream JTC as soon as the scaling feature got merged there.